wellcomecollection/catalogue-pipeline
# Local relation embedder

## What does this change?

This adds a new Main object - CLIMain, which can be invoked on the command line to run the relation embedder
resolves  https://github.com/wellcomecollection/catalogue-pipeline/issues/2782
## How to test

With an appropriate AWS_PROFILE set, 

```
% sbt "project relation_embedder" ";stage"
% cat pipeline/relation_embedder/relation_embedder/scripts/batches.txt | python pipeline/relation_embedder/relation_embedder/scripts/cli.py 2024-11-18      
```

It should now print out a load of Json, some log entries, and all the work ids that the real relation embedder would be passing downstream.

## How can we measure success?

This makes it easier to be confident of any changes we make to the embedder by allowing us to run local code against live upstream data. 

This is particularly important for the relation embedder over the other pipeline stages, as the behaviour of this stage is most dependent on the data in the upstream database, as it performs a search to find the relations.

## Have we considered potential risks?

Access is granted on a readonly basis.

We recently had to change the existing ECS task definitions manually to account for the lack of CLIMain.  We will have to revert those changes back to the properly deployed version.
